### ISSUE TITLE
Access CI history feature

### ISSUE THREAD

**Issue by amandakrohn** [ludwigjo/SE-Gorup10-CI#8]
Create a class (HistoryHandler) which is responsible for adding/accessing data from past and present CI builds and tests. 

**Requirements:**

- [x] #16 
- [x] #14 
- [x] #55

**Comment by BrorSebastianSjovald** [comment]
For the project structure I was thinking something like this:

```
SE-Gorup10-CI
│   README.md
│   pom.xml
│   run_ngrok.sh
│   .gitignore    
│
└───src
│   └───main/java/com/group10/CI
│   │  │   ContinuousIntegrationServer.java
│   │  │   etc...
│   └───test/java/com/group10/CI   
│       │   ContinuousIntegrationServerTest.java
│       │   etc...
│
└───History
    │   historyFile1.json
    │   historyFile2.json
```

My idea was to save the history locally (no need for consistency between different servers) in a json file. We do this in a  History folder which we gitignore. This way it is easy to structure info like if it failed, got an error or succeeded. For example it could look something like this:

```json
{
  "PR info?": {
      "author": "x"
      "time": "y"
      "etc": "etc"
  }
  "status": "success"
  "build": "build console output"
  "test": "test console output"
}
``` 

Does this sound like a good idea? Do you have any other ideas?

**Comment by ludwigjo** [comment]
I like the structure and a JSON based storage seems nice!

**Comment by darkway-s** [comment]
I agree with this structure. Storing the History "locally" is actually submitting it to the GITHUB server. To assure the file of every history is unique, we may generate a file name with something like a timestamp.

**Comment by BrorSebastianSjovald** [comment]
It is only submitted to GitHub if we don't include it in the .gitignore file. Do you think we should upload the History folder? Optionally I can create a simple non-relational DB using Firebase to store the data. It shouldn't take long

**Comment by BrorSebastianSjovald** [comment]
I will assign myself to this if anyone doesn't mind

**Comment by BrorSebastianSjovald** [comment]
Closed finally by #58
